build_scripts/generate-manifest-helpers.js

```diff
@@ -37,6 +37,33 @@ function writeDataToFile(file, data) {
   }
 }
 
+function getCurrentGitRevision() {
+  if (process.env.GIT_REVISION) {
+    // If no hash was passed, attempt to read it from disk
+    return process.env.GIT_REVISION
+  }
+  try {
+    const rev = fs
+      .readFileSync('.git/HEAD')
+      .toString()
+      .trim()
+      .split(/.*[: ]/)
+      .slice(-1)[0]
+    if (rev.indexOf('/') === -1) {
+      return rev
+    } else {
+      return fs
+        .readFileSync('.git/' + rev)
+        .toString()
+        .trim()
+    }
```

Comment:
Please add a comment or link to inspiration for this and remove support for detached head which I don't think we need, to simplify things.